backend/internal/api/handlers.go

```diff
@@ -245,6 +245,35 @@ func (h *Handlers) GetSMTPConfig(c echo.Context) error {
 	return c.JSON(http.StatusOK, response)
 }
 
+// MarkVideoAsWatched handles POST /api/videos/:videoId/watch
+func (h *Handlers) MarkVideoAsWatched(c echo.Context) error {
+	videoID := c.Param("videoId")
+	if videoID == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "Video ID is required")
+	}
+
+	// Basic validation for video ID format (e.g., "yt:video:VIDEO_ID")
+	// This should align with the OpenAPI spec pattern.
+	if !strings.HasPrefix(videoID, "yt:video:") || len(videoID) < 19 { // "yt:video:" is 9 chars + 11 for ID = 20. Let's use 19 as a loose minimum.
```

Comment:
I think there's some duplicated checking for this - can we create a standard YT ID validation?